randovania/game_connection/connector/am2r_remote_connector.py

```diff
@@ -0,0 +1,241 @@
+import logging
+import uuid
+from typing import TYPE_CHECKING
+
+from qasync import asyncSlot
+
+from randovania.game_connection.connector.remote_connector import (
+    PickupEntryWithOwner,
+    PlayerLocationEvent,
+    RemoteConnector,
+)
+from randovania.game_connection.executor.am2r_executor import AM2RExecutor
+from randovania.game_description import default_database
+from randovania.game_description.db.region import Region
+from randovania.game_description.resources.inventory import Inventory, InventoryItem
+from randovania.game_description.resources.pickup_index import PickupIndex
+from randovania.games.game import RandovaniaGame
+
+if TYPE_CHECKING:
+    from randovania.game_description.pickup.pickup_entry import PickupEntry
+    from randovania.game_description.resources.item_resource_info import ItemResourceInfo
+
+
+class AM2RRemoteConnector(RemoteConnector):
+    _game_enum: RandovaniaGame = RandovaniaGame.AM2R
+
+    def __init__(self, executor: AM2RExecutor):
+        super().__init__()
+        self._layout_uuid = uuid.UUID(executor.layout_uuid_str)
+        self.logger = logging.getLogger(type(self).__name__)
+        self.executor = executor
+        self.game = default_database.game_description_for(RandovaniaGame.AM2R)
+
+        self.reset_values()
+
+        self.executor.signals.new_inventory.connect(self.new_inventory_received)
+        self.executor.signals.new_collected_locations.connect(self.new_collected_locations_received)
+        self.executor.signals.new_player_location.connect(self.new_player_location_received)
+        self.executor.signals.new_received_pickups.connect(self.new_received_pickups_received)
+        self.executor.signals.connection_lost.connect(self.connection_lost)
+
+    @property
+    def game_enum(self) -> RandovaniaGame:
+        return self._game_enum
+
+    def description(self) -> str:
+        return f"{self.game_enum.long_name}"
+
+    async def current_game_status(self) -> tuple[bool, Region | None]:
+        return (self.in_cooldown, self.current_region)
+
+    def connection_lost(self) -> None:
+        self.logger.info("Finishing connector")
+        # TODO: Finished signal is never used. Remove it everywhere?
+        self.Finished.emit()
```

Comment:
Reminds me that we should delete it if it is still not used after all these weeks since the multiworld rewrite.